### PR TITLE
Prevent possible undo delete group by flow

### PR DESCRIPTION
**Pull request by norkans7**
FIxes released group that craches the flow editor, api flow list

- https://app.getsentry.com/nyaruka/textit/issues/112091636/
- https://app.getsentry.com/nyaruka/textit/issues/112090513/
- https://app.getsentry.com/nyaruka/textit/issues/112090514/

### CODE REVIEW

File: temba/api/v2/views.py [hunk 1]
```diff
@@ -224,7 +224,7 @@ def filter_queryset(self, queryset):
         # filter by group name/uuid (optional)
         group_ref = params.get('group')
         if group_ref:
-            group = ContactGroup.user_groups.filter(org=org, is_active=True).filter(Q(uuid=group_ref) | Q(name=group_ref)).first()
```
Comment (nicpottier) [inline temba/api/v2/views.py]: Not sure I understand these changes? Why wouldn't we say the uuid is invalid if the group has been deleted?

Comment (nicpottier) [inline temba/api/v2/views.py]: Oh nm.. see the change to the manager below..

File: temba/contacts/models.py [hunk 2]
```diff
@@ -1563,15 +1564,21 @@ def get_user_group(cls, org, name):
...
+        existing = ContactGroup.user_groups.filter(org=org, id=group_id).first()
+        if not existing:
+            existing = ContactGroup.get_user_group(org, name)
+
         if existing:
             return existing
-        else:
-            return cls.create(org, user, name)
+
+        if name is None:
```
Comment (nicpottier) [inline temba/contacts/models.py]: I don't think we should have ```get_or_create``` methods that can return None. get_or_create should always return something, if you want to add a lookup by id or whatnot then that's ok, but we should keep the contract here simple and clear.

Comment (norkans7) [inline temba/contacts/models.py]: Right, the cause of the empty name also is fixed; so no case for name=None, I remove this now

File: temba/contacts/models.py [hunk 3]
```diff
@@ -1563,15 +1564,18 @@ def get_user_group(cls, org, name):
...
         Returns the user group with the passed in name
         """
-        return ContactGroup.user_groups.filter(name__iexact=cls.clean_name(name), org=org, is_active=True).first()
+        return ContactGroup.user_groups.filter(name__iexact=cls.clean_name(name), org=org).first()
 
     @classmethod
-    def get_or_create(cls, org, user, name):
-        existing = ContactGroup.get_user_group(org, name)
+    def get_or_create(cls, org, user, name, group_id=None):
+        existing = ContactGroup.user_groups.filter(org=org, id=group_id).first()
```
Comment (nicpottier) [inline temba/contacts/models.py]: Doesn't seem like we should be querying using group_id if it is an optional argument. At least we should be testing against None before doing so.

File: temba/flows/models.py [hunk 4]
```diff
@@ -4011,18 +4011,8 @@ def get_groups(cls, org, json_obj):
                 except Exception:
                     group_id = -1
```
Comment (nicpottier) [inline temba/flows/models.py]: group_id is being set to -1 here, but in get_or_create we are testing against group_id being None. Probably want those to be in sync.

**Comment by nicpottier** [comment]
Woot, nice!

**Pull request merged**